### PR TITLE
Add a partition for projects in the slurm acct sync (HPC-10636)

### PR DESCRIPTION
**Pull request by itkovian**

### CODE REVIEW

File: lib/vsc/administration/slurm/sync.py [hunk 1]
```diff
@@ -236,29 +237,51 @@ def slurm_project_users_accounts(
 
     for cluster in clusters:
         cluster_users_acct = [
-            (user.User, user.Account) for user in slurm_user_info if user and user.Cluster == cluster
+            (user.User, user.Account, user.Partition) for user in slurm_user_info if user and user.Cluster == cluster
         ]
 
-        protected_users = [u for (u, a) in cluster_users_acct if a in protected_accounts]
+        print(f"cluster_users_acct: {cluster_users_acct}")
```
Comment (stdweird) [inline lib/vsc/administration/slurm/sync.py:243]: no print please

File: lib/vsc/administration/slurm/sync.py [hunk 2]
```diff
@@ -236,29 +237,49 @@ def slurm_project_users_accounts(
 
     for cluster in clusters:
         cluster_users_acct = [
-            (user.User, user.Account) for user in slurm_user_info if user and user.Cluster == cluster
+            (user.User, user.Account, user.Partition) for user in slurm_user_info if user and user.Cluster == cluster
```
[the remark this answers is not in the document]
Comment (itkovian) [reply lib/vsc/administration/slurm/sync.py:240]: A tuple. A list is not hashable ;p

[the remark this answers is not in the document]
Comment (stdweird) [reply lib/vsc/administration/slurm/sync.py:240]: if it is alwasy a tuple, i prefer `Partitions` as a name. idem with some of the variables below. it's not a blocker for me if it is too much work to change

File: lib/vsc/administration/slurm/sync.py [hunk 3]
```diff
@@ -236,60 +237,91 @@ def slurm_project_users_accounts(
...
         # create associations in the default account for users that do not already have one
-        cluster_users_with_default_account = set([u for (u, a) in cluster_users_acct if a == default_account])
         commands.extend([create_add_user_command(
             user=user,
             account=default_account,
             default_account=default_account,
-            cluster=cluster) for (user, _) in new_users if user not in cluster_users_with_default_account
+            cluster=cluster,
+            partition=p)
+            for (user, _, project_partition) in new_users for p in project_partition
```
Comment (stdweird) [inline lib/vsc/administration/slurm/sync.py:293]: rename `project_partition` -> project_partitions

File: lib/vsc/administration/slurm/sync.py [hunk 4]
```diff
@@ -236,60 +237,91 @@ def slurm_project_users_accounts(
...
+
+        # kick out users no longer in the project or whose partition changed
         commands.extend([
             create_remove_user_account_command(user=user, account=project_name, cluster=cluster)
             for (user, project_name) in remove_project_users
         ])
 
+        commands.extend([
+            create_remove_user_account_command(user=user, account=project_name, cluster=cluster, partition=partition)
+            for (user, project_name, partition) in obsolete_slurm_project_users
```
Comment (stdweird) [inline lib/vsc/administration/slurm/sync.py:322]: is partition here a list or single partition?

**Pull request merged**